python/activity_stream/widget_note.py

```diff
@@ -383,16 +480,79 @@ def set_note_info(self, data):
         # higher level than this widget, but we're still going to be
         # careful here, because we saw this bug crop up during Cut Support
         # QA.
-        self.ui.content.setText(data.get("content", ""))
+        self._set_note_content_text(data.get("content", ""))
+        # The underlying text document's line count is not showing the correct
+        # number immediately after setting the document's text directly. Trigger a refresh
+        # of the widget size with the updated correct line count.
+        QtCore.QTimer.singleShot(0, self._update_note_content_size)
 
         # This allows selections from higher-level layouts to occur. If
         # we don't set this, the label accepts and blocks mouse click
         # events, which means if you expect to select the note widget
         # itself you can't click on the note contents.
-        self.ui.content.setAttribute(QtCore.Qt.WA_TransparentForMouseEvents)
-        
+        self.ui.content_editable.setAttribute(QtCore.Qt.WA_TransparentForMouseEvents)
+
         # format note links
         if self.show_note_links:
             html_link_box_data = data.get("note_links", []) + data.get("tasks", [])
             links_html = self.__generate_note_links_table(html_link_box_data)
             self.ui.links.setText(links_html)
+            self.ui.links.show()
+
+    ##########################################################################
+    # internal utilities
+
+    def _set_note_content_text(self, text):
+        self._saved_content = text
+        self.ui.content_editable.document().setPlainText(text)
+
+    def _update_note_content_size(self):
+        widget = self.ui.content_editable
+        viewable_lines = widget.document().lineCount()
+        max_visible_lines = 6
```

Comment:
please convert into a `MAX_VISIBLE_LINES` constant and place at the top of the class.